### PR TITLE
MM-15498 - removed redundant post_edited event on reaction change

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
When publishing reaction change event (add/remove) there was a redundant call to publish post_edited event that was possibly causing an OOM on channels with large amount of users.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-15498

### CODE REVIEW

File: app/reaction.go [hunk 1]
```diff
@@ -136,10 +136,4 @@ func (a *App) sendReactionEvent(event string, reaction *model.Reaction, post *mo
 
 	post.HasReactions = hasReactions
 	post.UpdateAt = model.GetMillis()
-
```
Comment (saturninoabril) [inline app/reaction.go]: If lines below are removed, 2 lines above can be removed as well.

Comment (reflog) [reply app/reaction.go]: You are right. Will clean it up

**Comment by reflog** [comment]
> (1) I'm curious if you have investigation result that supported this change. Maybe good to add in the PR description.
> (2) Also, does this change require client (webapp/mobile) change?

It's based on @jwilander investigation into the issue and right now I am working on getting load-test working to see if we actually got a memory usage improvement.

It doesn't require a change in webapp/mobile.

**Comment by jasonblais** [comment]
@jwilander Ready for your review when you have the chance

@reflog may need rebase prior to merge

**Comment by jwilander** [comment]
Sorry not sure how I missed that

**Comment by reflog** [comment]
So, finally I've managed to run two loadtest sessions to compare the hows this PR affects memory usage.
I'm attaching pprof dumps to this comment.
A '--alloc_space' comparision seems to prove that the situation improved, though I'd appreciate another set of eyes on the pprof analysis.

```
❯ go tool pprof --alloc_space --base prof/heap prof2/heap
File: mattermost
Type: alloc_space
Time: Jun 16, 2019 at 2:05pm (IDT)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) web
[74045:74062:0616/141349.360481:ERROR:browser_process_sub_thread.cc(221)] Waited 6 ms for network service
(pprof) top
Showing nodes accounting for -1.14GB, 9.95% of 11.40GB total
Dropped 902 nodes (cum <= 0.06GB)
Showing top 10 nodes out of 331
      flat  flat%   sum%        cum   cum%
   -0.55GB  4.87%  4.87%    -1.10GB  9.69%  github.com/mattermost/mattermost-server/app.(*WebConn).writePump
   -0.53GB  4.64%  9.51%    -0.52GB  4.60%  fmt.Sprintf
   -0.11GB  0.97% 10.48%    -0.15GB  1.34%  compress/flate.NewWriter
   -0.06GB  0.55% 11.03%    -0.04GB  0.39%  encoding/json.Marshal
    0.06GB   0.5% 10.52%     0.06GB   0.5%  strings.(*Builder).grow
    0.06GB  0.49% 10.03%     0.06GB  0.49%  strings.genSplit
    0.04GB  0.38%  9.65%     0.04GB  0.38%  reflect.unsafe_NewArray
   -0.04GB  0.32%  9.97%    -0.04GB  0.32%  compress/flate.(*compressor).initDeflate
    0.04GB  0.31%  9.65%     0.04GB  0.31%  github.com/mattermost/mattermost-server/vendor/go.uber.org/zap/buffer.NewPool.func1
   -0.03GB   0.3%  9.95%    -0.07GB  0.64%  github.com/mattermost/mattermost-server/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather

```
[pprof.tar.gz](https://github.com/mattermost/mattermost-server/files/3294061/pprof.tar.gz)

**Comment by jwilander** [comment]
Looks good 👍 

Curious, were you running with default settings for the load test? I imagine if you tweaked `LoadtestEnviromentConfig. PercentUsersHighVolumeChannel` to be higher (it controls the percentage of the total users to put into high volume channels) you would see a bigger gap between the base and the changes.

**Comment by reflog** [comment]
@jwilander - yeah, all default, except for NumEmojis (100).
Is this ready enough for the merge?

**Pull request merged**